arrays/arrays.c

```diff
@@ -83,14 +99,22 @@ char *arr_read(Array *arr, int index) {
 void arr_insert(Array *arr, char *element, int index) {
 
   // Throw an error if the index is greater than the current count
-
+  if (index > arr->count) {
+    fprintf(stderr, "INSERT error: %d", index);
+  }
   // Resize the array if the number of elements is over capacity
-
+  if (arr->count + 1 > arr->capacity) { 
+    resize_array(arr);
+  }
   // Move every element after the insert index to the right one position
-
+  for(int i = index; i < arr->count; i++) {
```

Comment:
I see what you're doing here, it would be easier to loop from the (right) end of the array (i.e. from int i = arr->count; i > index; i--) instead of looping from the middle of the array to the end as you have here. When you do it it this way you're susceptible to duplicating values, especially your end value. You can see it if you change the code on line 115 first. If you write down the loop's pattern on paper in either direction you'll see the drawback to looping "up" as you do here.